libs/hdf-converters/test/attestations/attestations.spec.ts

```diff
@@ -6,108 +6,371 @@ import {
-      expect.stringMatching(/^Attestation/)
-    );
     expect(expiredAttestationMessage).toEqual(
       expect.stringMatching(/^Expired/)
     );
   });
+});
+
+describe('convertAttestationToSegment', () => {
+  it('Should convert a passing and unexpired attestation message to an HDF file segment', () => {
+    const segment_unexpired_pass = convertAttestationToSegment(
+      attestationMessageArray[0]
+    );
+    expect(segment_unexpired_pass.status).toEqual(
+      ExecJSON.ControlResultStatus.Passed
+    );
+    expect(segment_unexpired_pass.message).toEqual(
+      createAttestationMessage(attestationMessageArray[0], false)
+    );
+  });
+
+  it('Should convert a passing and expired attestation message to an HDF file segment', () => {
+    const segment_expired_pass = convertAttestationToSegment(
+      attestationMessageArray[1]
+    );
+    expect(segment_expired_pass.status).toEqual(
+      ExecJSON.ControlResultStatus.Skipped
+    );
+    expect(segment_expired_pass.message).toEqual(
+      createAttestationMessage(attestationMessageArray[1], true)
+    );
+  });
+
+  it('Should convert a failing and unexpired attestation message to an HDF file segment', () => {
+    const segment_unexpired_fail = convertAttestationToSegment(
+      attestationMessageArray[2]
+    );
+    expect(segment_unexpired_fail.status).toEqual(
+      ExecJSON.ControlResultStatus.Failed
+    );
+    expect(segment_unexpired_fail.message).toEqual(
+      createAttestationMessage(attestationMessageArray[2], false)
+    );
+  });
 
-  it('Should successfully add an attestation to a results set and ignore expired ones', () => {
-    const inputData = JSON.parse(
+  it('Should convert a failing and expired attestation message to an HDF file segment', () => {
+    const segment_expired_fail = convertAttestationToSegment(
+      attestationMessageArray[3]
+    );
+    expect(segment_expired_fail.status).toEqual(
+      ExecJSON.ControlResultStatus.Skipped
+    );
+    expect(segment_expired_fail.message).toEqual(
+      createAttestationMessage(attestationMessageArray[3], true)
+    );
+  });
+});
+
+describe('addAttestationToHDF', () => {
+  let inputData: ExecJSON.Execution;
+  const consoleOriginal = console.error;
+  // Reset inputData for each test
+  beforeEach(() => {
+    inputData = JSON.parse(
       fs.readFileSync(
-        'sample_jsons/attestations/sample_input_report/rhel7-results.json',
+        'sample_jsons/attestations/sample_input_report/rhel8_sample_more.json',
         'utf-8'
       )
     ) as ExecJSON.Execution;
 
-    const output = addAttestationToHDF(inputData, attestationArray);
+    console.error = jest.fn();
+  });
+
+  afterEach(() => {
+    console.error = consoleOriginal;
+  });
 
-    //fs.writeFileSync('sample_jsons/attestations/rhel7-json-attestedjson.json', JSON.stringify(output, null, 2))
+  /**
+     * Adding an attestation results in HDF object having an additional result object in controls.results array
```

Comment:
```suggestion
     * Adding an attestation creates an additional result object in controls.results array in the HDF object 
```

Using a word like `results` here as a verb gets confusing because we are also using it so much as a noun, so I would just restate to something like this.